model/host/ip_address.go

```diff
@@ -0,0 +1,21 @@
+package host
+
+import (
+	"context"
+
+	"github.com/evergreen-ci/evergreen/db"
+)
+
+// IPAddress represents a single public IPv4 address available to use for a
+// host.
+type IPAddress struct {
+	ID string `bson:"_id"`
```

Comment:
I noticed we use an ObjectID, but then you convert it to Hex. Could we just store it as an ObjectID here too? Is there any reason not to?

Reply:
That makes a lot of sense, I'm good with using strings here